### ISSUE TITLE
resourceLabelFormatters include a leading backslash on Windows paths and stripPathStartingSeparator doesn't affect this

### ISSUE THREAD

**Issue by DanTup** [microsoft/vscode#202678]
If I do _not_ specify a `resourceLabelFormatter` for a scheme, and the `path` component of that scheme looks like a file path, VS Code renders it as the file path:

Actual URI: `my-custom-file:///C::/Dev/Test%20Projects/macro-sample/bin/danny.dart`

Display:

![image](https://github.com/microsoft/vscode/assets/1078012/8cc0afc7-8bf0-4462-ba15-34549c8ce1df)

If I want to show some suffix on the end of this, I can use the `formatting.label` field:

```js
"resourceLabelFormatters": [
	{
		"scheme": "my-custom-file",
		"formatting": {
			"label": "${path} (generated file)"
		}
	}
],
```

However, the `${path}` here includes the leading slash from the path that VS Code was previously removing:

![image](https://github.com/microsoft/vscode/assets/1078012/a451ae26-f0e1-4611-ab3b-116a8db533f4)

This feels inconsistent - it should be possible to use `label` but include the same path that would be shown without it.

`stripPathStartingSeparator` looks like it might be the answer, but a) it doesn't make any difference in my testing and b) it's not clear if that would also strip a leading `/` on non-Windows platforms.

**Comment by alexandruavadanii** [comment]
I don't see this problem with my Uris and I use something like:
```
"scheme": "cc",
        "formatting": {
          "label": "${path} (${query.pv})",
          "separator": "/",
          "workspaceSuffix": "ClearCase"
        }
```
You could also write your own logic for parsing the path and injecting that in `Uri.query` ...

**Comment by DanTup** [comment]
@alexandruavadanii seems like you're explicitly using `/` though, and my goal here was to show the Windows path (eg. using backslashes but without the leading backslash)?

> You could also write your own logic for parsing the path and injecting that in Uri.query ...

The URIs here are generated by an LSP server and are not VS Code-specific. I don't want to change the underlying URIs just to improve VS Code's rendering - my understanding is that `resourceLabelFormatters` are supposed to be the way to control this formatting, it just feels like they have some quirks.

**Comment by alexandruavadanii** [comment]
@DanTup true, I only use `/` and even that way I ran into issues (e.g. my `query.pv` might also contain `/` in which case vscode will mess up the short name displayed ... I ended up using an unicode chat that resembles `/` to bypass that).

Though, is `\C:\something` a common path format? As far as I read, paths starting with `\` on Windows should be treated as relative, so resolving that path might yield unexpected results.

Anyway, does this help?
https://github.com/gitkraken/vscode-gitlens/blob/main/package.json#L15824C7-L15824C27

**Comment by DanTup** [comment]
> Though, is \C:\something a common path format? As far as I read, paths starting with \ on Windows should be treated as relative, so resolving that path might yield unexpected results.

To be clear, I _don't_ want that leading backslash :)

If I do not provide a `label` at all, VS Code shows a valid Windows file path (even though the URI is not a `file://` URI but another scheme that happens to work just like `file:`). However, if I want to use `label` (eg. to add a suffix), there is no way (that I can find) to include a `${path}` that is rendered the same way as if I didn't have a label.

So if I want "the default  behaviour without a label" but just with some additional suffix/prefix added to the label, I can't currently get that.


> Anyway, does this help?

As far as I can tell, that just controls lowercasing of the drive letter?

**Comment by alexandruavadanii** [comment]
> As far as I can tell, that just controls lowercasing of the drive letter?
Sorry, I think you're right.
I saw `path.splice(1) in [the implementation here](https://github.com/microsoft/vscode/blob/fa989a17f1985a641e8338229e7d8ce6980a4d5c/src/vs/base/common/labels.ts#L143) and failed to compute that the first char is still added ...
